test/runtime/fqdn.go

```diff
@@ -778,14 +782,94 @@ INITSYSTEM=SYSTEMD`
 		})
 	})
 
+	It("DNS proxy policy works if Cilium stops", func() {
+		targetURL := "http://world1.cilium.test"
+		targetIP := worldIps[WorldHttpd1]
+		invalidURL := "http://world1.outside.test"
```

Comment:
This is fine, but is there a reason to not use `world1.cilium.test` and `world2.cilium.test`? the policy is specific to the test, so the L7 doesn't need a wildcard `"matchPattern": "*.cilium.test"`

Reply:
just because is easy to read that the invalid is in a different domain.